testharness/src/main/java/com/google/accompanist/testharness/TestHarness.kt

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * Render [content] in a [Box] within a harness, overriding various device configuration values to
+ * make testing easier.
+ *
+ * @param size if not [DpSize.Unspecified], the [content] will be forced to be drawn with at this
+ * size, overriding [LocalDensity] if necessary to ensure that there is enough space. This
+ * defaults to [DpSize.Unspecified].
+ *
+ * @param darkMode if true, the content will be rendered with dark mode. This defaults to the
+ * current dark mode value as reported by [isSystemInDarkTheme].
+ *
+ * @param locales the list of locales to render the app with. This defaults to the list of locales
+ * returned by [LocalConfiguration.current].
+ *
+ * @param layoutDirection an overriding layout direction. This defaults to `null`, which means
+ * that the layout direction from the [locales] is used instead.
+ *
+ * @param fontScale the font scale to render text at. This defaults to the current
+ * [Density.fontScale].
+ *
+ * @param fontWeightAdjustment the font weight adjustment for fonts. This defaults to the current
+ * [fontWeightAdjustment] (if any). If `null`, the [fontWeightAdjustment] will be left unchanged.
+ */
+@Composable
+fun TestHarness(
+    size: DpSize = DpSize.Unspecified,
+    darkMode: Boolean = isSystemInDarkTheme(),
+    locales: LocaleListCompat = ConfigurationCompat.getLocales(LocalConfiguration.current),
+    layoutDirection: LayoutDirection? = null,
+    fontScale: Float = LocalDensity.current.fontScale,
+    fontWeightAdjustment: Int? =
+        if (Build.VERSION.SDK_INT >= 31) LocalConfiguration.current.fontWeightAdjustment else null,
+    content: @Composable () -> Unit
+) {
+    // Use the DensityForcedSize content wrapper if specified
+    val sizeContentWrapper: @Composable (@Composable () -> Unit) -> Unit =
+        if (size == DpSize.Unspecified) {
+            { it() }
+        } else {
+            { DensityForcedSize(size, it) }
+        }
+
+    // First override the density. Doing this first allows using the resulting density in the
+    // overridden configuration.
+    sizeContentWrapper {
+        // Second, override the configuration, with the current configuration modified by the
+        // given parameters.
+        OverriddenConfiguration(
+            configuration = Configuration().apply {
+                // Initialize from the current configuration
+                updateFrom(LocalConfiguration.current)
+                // Set dark mode directly
+                uiMode = uiMode and Configuration.UI_MODE_NIGHT_MASK.inv() or if (darkMode) {
+                    Configuration.UI_MODE_NIGHT_YES
+                } else {
+                    Configuration.UI_MODE_NIGHT_NO
+                }
+                // Update the locale list
+                if (Build.VERSION.SDK_INT >= 24) {
+                    setLocales(LocaleList.forLanguageTags(locales.toLanguageTags()))
+                } else {
+                    setLocale(locales[0])
+                }
```

Comment:
@alexvanyo `java.util.Locale.setDefault(locales[0])` would be also useful here since java formatters won't know about the locale update otherwise

Reply:
I'd be interested to hear your use case here, since I'd intentionally prefer to avoid calling `java.util.Locale.setDefault(locales[0])` if possible.

`java.util.Locale.getDefault()` isn't observable by Compose, so if you're calling that from a `@Composable` function, you might not be notified if the locale changes in the general case (for instance, if opting out of activity recreation for locale changes)

If possible, can you retrieve the locale from `LocalConfiguration.current.locales` instead to create the formatter?